rfc/system/4358-tuple-idl.md

```diff
@@ -0,0 +1,562 @@
+# Tuple IDL
+
+**Authors:**
+
+- [@Chun-Mao (Michael) Lai](https://github.com/Mecoli1219)
+
+## 1 Executive Summary
+
+Goals:
+
+- Implement support for tuple types (lists of different types) in FlyteIDL.
+    - Example: a : Tuple[int, str, float] must be assigned an integer for the first element, a string for the second element, and a floating value for the third element.
+- Implement support for new Python types in Flytekit:
+    - `typing.NamedTuple`
+        - `typing.NamedTuple("NAME", ("K1", T1), ("K2", T2))`
+    - `typing.Tuple`
+        - `typing.Tuple[T1, T2, T3]` (Simplified case for NamedTuple)
+        - `typing.Tuple[T1, ...]` (It is quite similar to `typing.List[T1]`, we can address it in this RFC or use `LiteralType_CollectionType` to support it for now.)
+
+Implementation:
```

Comment:
Nit: could you please move this further below? I feel this is too much detail for the executive summary.

Reply:
Let's move this to `## 3 Proposed Implementation`.